packages/stripe-integration/src/stripe-upe/stripe-upe-integration-service.ts

```diff
@@ -0,0 +1,222 @@
+
+                if (!paymentElement) {
+                    return;
+                }
+
+                let error;
+
+                try {
+                    await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
+                        params: { method: methodId },
+                    });
+                } catch (err) {
+                    error = err;
+                }
+
+                if (error && error instanceof Error) {
+                    if (this.isMounted) {
+                        paymentElement.unmount();
+                        this.isMounted = false;
+                    }
+
+                    stripeupe.onError?.(error);
+                } else if (!this.isMounted) {
+                    await stripeElements?.fetchUpdates();
+                    this.mountElement(paymentElement, stripeupe.containerId);
+                }
+            },
+            (state) => state.getCheckout()?.outstandingBalance,
+            (state) => state.getCheckout()?.coupons,
+        );
+    }
+
+    mountElement(stripeElement: StripeElement, containerId: string): void {
+        if (!document.getElementById(containerId)) {
+            return;
+        }
+
+        stripeElement.mount(`#${containerId}`);
+        this.isMounted = true;
+    }
+
+    mapAppearanceVariables(styles: { [key: string]: string }) {
+        return {
+            colorPrimary: styles.fieldInnerShadow,
+            colorBackground: styles.fieldBackground,
+            colorText: styles.labelText,
+            colorDanger: styles.fieldErrorText,
+            colorTextSecondary: styles.labelText,
+            colorTextPlaceholder: styles.fieldPlaceholderText,
+            colorIcon: styles.fieldPlaceholderText,
+        };
+    }
+
+    mapInputAppearanceRules(styles: { [key: string]: string }) {
+        return {
+            borderColor: styles.fieldBorder,
+            color: styles.fieldText,
+            boxShadow: styles.fieldInnerShadow,
+        };
+    }
+
+    throwDisplayableStripeError(stripeError: StripeError) {
+        if (
+            includes(['card_error', 'invalid_request_error', 'validation_error'], stripeError.type)
+        ) {
+            throw new Error(stripeError.message);
+        }
+    }
+
+    throwPaymentConfirmationProceedMessage() {
+        // INFO: for case if payment was successfully confirmed on Stripe side but on BC side something go wrong, request failed and order status hasn't changed yet
+        // For shopper we need to show additional message that BC is waiting for stripe confirmation, to prevent additional payment creation
+        throw new PaymentMethodFailedError(
+            "We've received your order and are processing your payment. Once the payment is verified, your order will be completed. We will send you an email when it's completed. Please note, this process may take a few minutes depending on the processing times of your chosen method.",
```

Comment:
We should figure out how to translate this message... As far as I know, it can be translated only on checkout-js project side

Reply:
yes, we should. But it is very edge case and for now such message was enough.
Let's do this in another task.

Reply:
Don't forget to create it please, since from my experience it is hard to hold everything in our heads when we develop large projects like this